pkg/cluster/cluster.go

```diff
@@ -329,6 +329,10 @@ func (c *Cluster) compareStatefulSetWith(statefulSet *appsv1.StatefulSet) *compa
 		match = false
 		reasons = append(reasons, "new statefulset's annotations doesn't match the current one")
 	}
+	if !reflect.DeepEqual(c.Statefulset.Spec.Template.Annotations, statefulSet.Spec.Template.Annotations) {
+		match = false
+		reasons = append(reasons, "new statefulset's pod's annotations don't match the current one")
+	}
```

Comment:
This can be removed as it is already checked a few lines [later](https://github.com/zalando/postgres-operator/blob/a337d3ca7ee19e8d5f9769c99715293a70fc2993/pkg/cluster/cluster.go#L383)

Reply:
@runyontr want to address this feedback? I think, it's redundant code.

Reply:
This check is for the annotations on the pod.  the one starting on line 328 is for the annotations on the statefulset.

Reply:
The check on line 328 is fine, but here (232) you are comparing the template which is already done in line 383 (what I have linked)